unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/SlitPlate.cs

```diff
@@ -0,0 +1,302 @@
+                                }
+                            }   
+                            else if (count == generatorCountPerSlit - 1)
+                            {
+                                waveGeneratorList[count + (slitIndex * (generatorCountPerSlit))].GetComponent<WaveGenerator>().SetPropagationMode(WaveGenerator.WavePropagation.Circular);
+                               
+                                if (generatorCountPerSlit > 2)
+                                {
+                                    waveGeneratorList[count + (slitIndex * (generatorCountPerSlit))].transform.localPosition = new Vector3(initialPositionRight - (generatorPlacementTransistion * 0.5f), left.transform.localPosition.y, left.transform.localPosition.z + 0.02f);
+                                }
+                            }
+                        }
+                    }                             
+                }
+            }
+            else
+            {
+                DeactivatePlateWaveGenerators();
+            }
+        }
+
+        private int CalculateGeneratorsPerSlit()
+        {        
+            int numberOfGeneratorsPerSlit = 1 + (int)((slitWidth * 10.0f)/1.5f);
+            return numberOfGeneratorsPerSlit; 
+        }
+
+        public void AddWaveGenerator()
+        {
+            var waveGenerator = new GameObject("WaveGenerator");
+            waveGenerator.transform.parent = GameObject.Find("Plate").transform;
+            waveGenerator.transform.rotation = waveGenerator.transform.parent.rotation;
+            //waveGenerator.transform.localPosition = left.transform.localPosition;
+
+            WaveGenerator waveGeneratorScript = waveGenerator.gameObject.AddComponent<WaveGenerator>();
+            waveGeneratorScript.WaveAmplitude = 0.5f;
+            waveGeneratorScript.WaveLength = 0.25f;
+            waveGeneratorScript.WaveFrequency = 0.5f;
+            waveGeneratorScript.SetPropagationMode(WaveGenerator.WavePropagation.Circular);
+            if (GameObject.Find("Water") != null)
+            {
+                waveGeneratorScript.SetPlaneObject(GameObject.Find("Water"));
+                waveGeneratorList.Add(waveGenerator);
+            }
+        }
+
+        public void AddAllWaveGenerators() 
+        {          
+            var totalNumberOfGenerators = generatorCountPerSlit * numberOfSlits; 
+            for (int count = 0; count < totalNumberOfGenerators - 1; count++)
+            {
+                AddWaveGenerator();
+            }
+            RegisterPlateWaveGenerators();
+        }
+
+        private void RegisterPlateWaveGenerators()
+        {
+            foreach (var generator in waveGeneratorList)

```

Comment:
We can simplify this to:
foreach (var generator in waveGeneratorList.Where(generator => generator != waveGenerator))